src/builder/create_components.rs

```diff
@@ -17,19 +16,20 @@ pub enum CreateActionRow<'a> {
     InputText(CreateInputText<'a>),
 }
 
-impl<'a> serde::Serialize for CreateActionRow<'a> {
+impl serde::Serialize for CreateActionRow<'_> {
     fn serialize<S: serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
-        use serde::ser::Error as _;
-
-        json!({
-            "type": 1,
-            "components": match self {
-                Self::Buttons(x) => serde_json::to_value(x).map_err(S::Error::custom)?,
-                Self::SelectMenu(x) => serde_json::to_value(vec![x]).map_err(S::Error::custom)?,
-                Self::InputText(x) => serde_json::to_value(vec![x]).map_err(S::Error::custom)?,
-            }
-        })
-        .serialize(serializer)
+        use serde::ser::SerializeMap as _;
+
+        let mut map = serializer.serialize_map(Some(2))?;
+        map.serialize_entry("kind", &1_u8)?;
```

Comment:
Shouldn't this be `type` instead of `kind`?

Reply:
Indeed, fixed.